plugins/spacelift/api_key.go

```diff
@@ -0,0 +1,101 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://docs.spacelift.io/integrations/api#spacelift-api-key-token"),
+		ManagementURL: sdk.URL("https://mycorp.app.spacelift.io/settings/api-keys"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Endpoint,
+				MarkdownDescription: "The URL to your Spacelift account, for example https://mycorp.app.spacelift.io",
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+					Prefix: "https://",
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "The ID of your Spacelift API key. Available via the Spacelift application.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
```

Comment:
Could we set a `Length` here? My API Key seems to be 26 characters long.

Reply:
@arunsathiya Is it for validation when importing a secret? I've set the `Length`, rebuilt the plugin, tried to import a new key and provided a single character as value for `APIKeyID`, no errors/warnings 🤔

Reply:
Yes, as well as potentially for auto-detecting secrets when importing this API credential into 1Password